flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -66,8 +70,52 @@ public Fabric8FlinkKubeClient(Configuration flinkConfig, KubernetesClient client
 	}
 
 	@Override
-	public void createTaskManagerPod(TaskManagerPodParameter parameter) {
-		// todo
+	public void createJobManagerComponent(KubernetesJobManagerSpecification kubernetesJMSpec) {
+		final Deployment deployment = kubernetesJMSpec.getDeployment();
+		final List<HasMetadata> accompanyingResources = kubernetesJMSpec.getAccompanyingResources();
+
+		// create Deployment
+		LOG.debug("Start to create deployment with spec {}", deployment.getSpec().toString());
+		final Deployment createdDeployment = this.internalClient
+			.apps()
+			.deployments()
+			.inNamespace(this.nameSpace)
+			.create(deployment);
+
+		// Note, we should use the server-side uid of the created Deployment for the OwnerReference.
+		setOwnerReference(createdDeployment, accompanyingResources);
```

Comment:
Since we change the owner reference here, the document needs to be updated.
https://ci.apache.org/projects/flink/flink-docs-master/ops/deployment/native_kubernetes.html#manual-resource-cleanup

Reply:
Fixed by [abfef13](https://github.com/apache/flink/pull/11233/commits/abfef134eb5940d7831dab4c794d654c2fa70263)

Reply:
The e2e test  `test_kubernetes_session.sh` also needs to be updated.

Reply:
Good catch. Fixed by [b088e0e](https://github.com/apache/flink/pull/11233/commits/b088e0e0530fa3178a547b26b7e950165181caaf)